conceptnet_sme/semantic_matching.py

```diff
@@ -513,7 +753,23 @@ def load_model(filename):
         total_accum.print("Total time to load model:")
         return model
 
-    def evaluate_conceptnet(self, dataset, cutoff_value=-1, output_filename=None):
+    def evaluate_edge(self, rel_idx, left_idx, right_idx):
```

Comment:
I'm guessing that this could be sped up a fair amount if it could evaluate a batch of edges instead of just one edge at a time, right?

Reply:
Based on earlier experience with make_edge_score_file, I thought the speed up would not be worth the extra code complexity.  But you motivated me to try it, and the speed gain is significant (when I tried before I think I only had one gpu, which might make a difference).  So now I've refactored both evaluate_conceptnet and evaluate_statistics to use score_edges (which works on batches).  That required some small changes to score_edges, and uncovered a bug in it (handling the last, short, batch could break on multiple gpu's) which I fixed.